packages/core/src/amazonqGumby/chat/controller/messenger/messenger.ts

```diff
@@ -0,0 +1,358 @@
+            new ChatMessage(
+                {
+                    message,
+                    messageType: 'ai-prompt',
+                    messageId: GumbyNamedMessages.COMPILATION_PROGRESS_MESSAGE,
+                },
+                tabID
+            )
+        )
+    }
+
+    public sendJobSubmittedMessage(tabID: string, disableJobActions: boolean = false) {
+        const message =
+            'Your job has been submitted for transformation. The code transformation process may take 10-30 mins depending on the size of your module. You can view the details in the transformation hub.'
+
+        const buttons: ChatItemButton[] = []
+
+        if (!disableJobActions) {
+            // Note: buttons can only be clicked once.
+            // To get around this, we remove the card after they're clicked and then
+            // resubmit the message.
+            buttons.push({
+                keepCardAfterClick: true,
+                text: 'Open Transformation Hub',
+                id: ButtonActions.VIEW_TRANSFORMATION_HUB,
+            })
+
+            buttons.push({
+                keepCardAfterClick: true,
+                text: 'Stop Transformation',
+                id: ButtonActions.STOP_TRANSFORMATION_JOB,
+            })
+        }
+
+        const jobSubmittedMessage = new ChatMessage(
+            {
+                message,
+                messageType: 'ai-prompt',
+                messageId: GumbyNamedMessages.JOB_SUBMISSION_STATUS_MESSAGE,
+                buttons,
+            },
+            tabID
+        )
+
+        this.dispatcher.sendChatMessage(jobSubmittedMessage)
+    }
+
+    public sendStaticTextResponse(type: StaticTextResponseType, tabID: string) {
+        let message = '...'
+
+        switch (type) {
+            case 'no-workspace-open':
+                message = 'To begin, please open a workspace.'
+                break
+            case 'no-project-found':
+            case 'no-java-project-found':
+                message =
+                    'None of your open projects are supported by Amazon Q Code Transformation. Currently, Amazon Q can only upgrade Java projects built on Maven. \n For more information, see the Amazon Q documentation.'
+                break
+            case 'no-maven-java-project-found':
+                message =
+                    'None of your open Java projects are supported by Amazon Q Code Transformation. Currently, Amazon Q can only upgrade Java projects built on Maven. A pom.xml must be present in the root of your project to upgrade it. \n For more information, see the Amazon Q documentation.'
+                break
+            case 'could-not-compile-project':
+                message = `Amazon Q couldn't execute the Maven install or Maven copy-dependencies commands. To troubleshoot, see the [Amazon Q documentation.](https://docs.aws.amazon.com/amazonq/latest/aws-builder-use-ug/troubleshooting-code-transformation.html#maven-commands-failing)`
```

Comment:
Maybe just generalize this to `Maven commands`

Reply:
I'll run this by Allie!